### DISCUSSION TITLE
How to find osgEarth debug library with CMake ?

### DISCUSSION THREAD

**Discussion by SPlesskin** [pelicanmapping/osgearth#2241] [Support Q&A]
Hi,

I've built and installed the latest version of osgEarth (3.3.0) with vcpkg. In my project, i use a CMakeLists.txt file to find the installed files (binaries, headers):

```
set(VCPKG_INSTALLED_DIR "${CMAKE_SOURCE_DIR}/vcpkg/installed/x64-windows")

list(APPEND CMAKE_PREFIX_PATH ${VCPKG_INSTALLED_DIR}
                              ${VCPKG_INSTALLED_DIR}/debug)

find_package(Qt5 COMPONENTS Widgets REQUIRED)

find_package(OpenSceneGraph REQUIRED COMPONENTS osg osgDB osgGA osgUtil osgViewer)
find_package(osgEarth CONFIG REQUIRED)
```

However, when i try to generate the makefile to compile my project in **debug mode**, CMake doesn't find the debug library (*.\vcpkg\installed\x64-windows\debug\lib\osgEarthd.lib*) but only the release library. What's missing in my CMakeLists.txt script to fix this issue? Thank you very much for your help.

**Comment by plevy** [comment]
Can you try building the latest osgEarth master?  The answer to your question is slightly more complicated as cmake is using the config file to locate the libraries.  The change is in [osgEarthConfig.cmake.in](https://github.com/gwaldron/osgearth/blob/master/osgEarthConfig.cmake.in)
which is the helper for find_package to locate the library.  Now you should see:
-- Found osgEarth: C:/osgearth/install2/lib/osgEarth.lib
-- osgEarth_FOUND: TRUE
-- osgEarth_INCLUDE_DIRS: C:/osgearth/install2/include
-- osgEarth_LIBRARIES: C:/osgearth/install2/lib/osgEarth.lib
-- osgEarth_LIBRARIES_DEBUG: C:/osgearth/install2/lib/osgEarthd.lib
-- osgEarth_LDFLAGS:
-- osgEarth_DEFINITIONS:
-- Configuring done
-- Generating done
